lib/doctest.js

```diff
@@ -308,9 +314,13 @@ function substring(input, start, end) {
   );
 }
 
-function wrap$js(test) {
-  var type = (esprima.parse (test[INPUT].value)).body[0].type;
-  return type === 'FunctionDeclaration' || type === 'VariableDeclaration' ?
+function wrap$js(test, sourceType) {
+  var type = (
+    esprima.parse (test[INPUT].value, {sourceType: sourceType})
+  ).body[0].type;
```

Comment:
This would read better, I think:

```javascript
var _ = esprima.parse (test[INPUT].value, {sourceType: sourceType});
var type = _.body[0].type;
```

Hopefully you can think of a better name than `_`. ;)

Reply:
`ast`?

Reply:
:zap:

Reply:
`ast` sounds good to me. :thumbsup: